Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -0,0 +1,2679 @@
+    return_outputs(human_readable, ec)
+
+
+def describe_volumes_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    obj = vars(client._client_config)
+    kwargs = {}
+    data = []
+
+    if args.get('filters') is not None:
+        kwargs.update({'Filters': parse_filter_field(args.get('filters'))})
+    if args.get('volumeIds') is not None:
+        kwargs.update({'VolumeIds': parse_resource_ids(args.get('volumeIds'))})
+
+    response = client.describe_volumes(**kwargs)
+
+    for i, volume in enumerate(response['Volumes']):
+        data.append({
+            'AvailabilityZone': volume['AvailabilityZone'],
+            'Encrypted': volume['Encrypted'],
+            'State': volume['State'],
+            'VolumeId': volume['VolumeId'],
+            'VolumeType': volume['VolumeType'],
+            'CreateTime': datetime.strftime(volume['CreateTime'], '%Y-%m-%dT%H:%M:%SZ'),
+        })
+        if 'Tags' in volume:
+            for tag in volume['Tags']:
+                data[i].update({
+                    tag['Key']: tag['Value']
+                })
+
+    output = json.dumps(response['Volumes'], cls=DatetimeEncoder)
+    raw = json.loads(output)
+    raw[0].update({'Region': obj['_user_provided_options']['region_name']})
+    ec = {'AWS.EC2.Volumes(val.VolumeId === obj.VolumeId)': raw}
+    human_readable = tableToMarkdown('AWS EC2 Volumes', data)
+    return_outputs(human_readable, ec)
+
+
+def describe_launch_templates_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    obj = vars(client._client_config)
+    kwargs = {}
+    data = []
+
+    if args.get('filters') is not None:
+        kwargs.update({'Filters': parse_filter_field(args.get('filters'))})
+    if args.get('launchTemplateIds') is not None:
+        kwargs.update({'LaunchTemplateIds': parse_resource_ids(args.get('launchTemplateIds'))})
+    if args.get('launchTemplateNames') is not None:
+        kwargs.update({'LaunchTemplateNames': parse_resource_ids(args.get('launchTemplateNamess'))})
+
+    response = client.describe_launch_templates(**kwargs)
+
+    for i, template in enumerate(response['LaunchTemplates']):
+        data.append({
+            'LaunchTemplateId': template['LaunchTemplateId'],
+            'LaunchTemplateName': template['LaunchTemplateName'],
+            'CreatedBy': template['CreatedBy'],
+            'DefaultVersionNumber': template['DefaultVersionNumber'],
+            'LatestVersionNumber': template['LatestVersionNumber'],
+            'CreateTime': datetime.strftime(template['CreateTime'], '%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
this can throw value error. better calculate it before the append and in a try block.